pkg/config/validation/validation.go

```diff
@@ -3208,6 +3208,11 @@ var ValidateServiceEntry = registerValidateFunc("ValidateServiceEntry",
 			if port.TargetPort != 0 {
 				errs = appendValidation(errs, ValidatePort(int(port.TargetPort)))
 			}
+			if len(serviceEntry.Addresses) == 0 {
+				if port.Protocol == "" || port.Protocol == "TCP" {
```

Comment:
Why TCP invalid?

Reply:
As explain in this comment https://github.com/istio/istio/pull/28664#issuecomment-725488225
When setting protocol to TCP with empty addresses, it makes all outgoing http traffic via port `xxxx` match `0.0.0.0_xxxx` tcp listener, which is unexpected.

Reply:
But 0.0.0.0_xxx would not happen when auto allocate ip for serviceentry.

Reply:
I agree with this solution since dns auto allocate is per proxy config